Source/drlg_l2.cpp

```diff
@@ -1677,15 +1685,13 @@ bool PlaceMiniSet(const BYTE *miniset, int tmin, int tmax, int cx, int cy, bool
 				sy = GenerateRnd(DMAXY - sh);
 				abort = false;
 			}
-			int ii = 2;
 
 			for (int yy = 0; yy < sh && abort; yy++) {
 				for (int xx = 0; xx < sw && abort; xx++) {
-					if (miniset[ii] != 0 && dungeon[xx + sx][yy + sy] != miniset[ii])
+					if (miniset.search[yy][xx] != 0 && dungeon[xx + sx][yy + sy] != miniset.search[yy][xx])
 						abort = false;
 					if (dflags[xx + sx][yy + sy] != 0)
 						abort = false;
-					ii++;
 				}
 			}

```

Comment:
This is the same logic as lines 1739-1746. Would be nice extracted as a function.